### DISCUSSION TITLE
petsc check error

### DISCUSSION THREAD

**Discussion by jin-psu** [idaholab/moose#19467] [Q&A Getting Started] [answered]
Dear all, 

We have been trying for multiple time to install moose on lemhi, 
The petsc installation process is fine (no error), but  "make PETSC_DIR=/your/home/projects/moose/scripts/../petsc PETSC_ARCH=arch-moose check" always has issue. The error message is attached.

It looks like mpi link issue, but the openmpi is loaded "openmpi/4.1.1-gcc-11.2.0-n2sm"

Really appreciate some guidance!

Cheers,
Miaomiao
[error.txt](https://github.com/idaholab/moose/files/7591033/error.txt)

**Comment by GiudGiud** [comment]
@fdkong openmpi on lemhi works fine for this?

**Comment by milljm** [comment, marked as answer]
I have been running into errors with gcc-11.2.0. You might want to try an older (stabler) variant.

I created a new stack due to this, it is available with:

```bash
$> module load use.moose civet_env
$> module list

Currently Loaded Modules:
  1) use.moose                       6) cmake/3.21.3-gcc-9.4.0-d22d
  2) flex/2.6.4-gcc-11.2.0-wvjp      7) patchelf/0.13-oneapi-2021.4.0-uxhq
  3) bison/3.7.6-gcc-11.2.0-aysl     8) miniforge
  4) gcc/9.4.0-gcc-8.4.1-57pg        9) civet_env
  5) mvapich2/2.3.6-gcc-9.4.0-n6hz
```
The above is what our Civet clients load, when we need to build PETSc.


Or if you would rather not build PETSc, you're welcome to use the default modules already in place:
```bash
$> module load use.moose PETSc
$> module list

Currently Loaded Modules:
  1) use.moose                     4) gcc/9.4.0-gcc-8.4.1-57pg
  2) flex/2.6.4-gcc-11.2.0-wvjp    5) mvapich2/2.3.6-gcc-9.4.0-n6hz
  3) bison/3.7.6-gcc-11.2.0-aysl   6) PETSc/3.15.1-gcc-9.4.0
```

Both methods though, should describe in enough detail what version of GCC we are using for the most part.

**Reply by jin-psu** [reply]
This works! Appreciate the help 👍
